src/projectile.cpp

```diff
@@ -0,0 +1,221 @@
+#include "projectile.h"
+
+#include "game.h"
+#include "map.h"
+#include "map_iterator.h"
+#include "explosion.h"
+#include "field.h"
+
+projectile::projectile() :
+    speed( 0 ), range( 0 ), drop( nullptr ), custom_explosion( nullptr )
+{ }
+
+projectile::~projectile()
```

Comment:
This could indeed be defaulted, as in `projectile::~projectile() = default;`. But it has to stay in the cpp file, not the header. If it was in the header, the header would have to include "item.h" as well to allow destroying the `item` inside of the `unique_ptr`. The current code allows using this class without having to include "item.h".